### ISSUE TITLE
UserSimulationDetailsCollectionResponse missing after 5.27.0-preview

### ISSUE THREAD

**Issue by cammurray** [microsoftgraph/msgraph-sdk-dotnet#2524]
### Describe the bug

The UserSimulationDetailsCollectionResponse class is missing in the C# library after 5.27.0-preview (5.27.0-preview is the last version that the class exists). This means that you cannot page UserSimulationDetails objects anymore.

### Expected behavior

UserSimulationDetailsCollectionResponse should exist to enable paging UserSimulationDetails objects.

### How to reproduce

This code works in 5.27.0-preview, but won't work after.

`
        var requestInformation =
            GraphClient.Security.AttackSimulation.Simulations[SimulationId].ToGetRequestInformation();

        requestInformation.URI = new Uri(requestInformation.URI.ToString() + "/report/simulationUsers");
        requestInformation.QueryParameters["Top"] = 1000;

        var results = await GraphClient.RequestAdapter.SendAsync<UserSimulationDetailsCollectionResponse>(requestInformation, UserSimulationDetailsCollectionResponse.CreateFromDiscriminatorValue);

        var pageIterator = Microsoft.Graph.PageIterator<UserSimulationDetails,UserSimulationDetailsCollectionResponse>
            .CreatePageIterator(GraphClient, results, async (userSimDetail) => {}
`

### SDK Version

5.28.0

### Latest version known to work for scenario above?

5.27.0

### Known Workarounds

Dumped the class out of the 5.27.0 library and make a local copy of it.

### Debug output

<details><summary>Click to expand log</summary>
```

<Log output here>

```
</details>

**Comment by andrueastman** [comment]
Thanks for raising this @cammurray 

Any chance you can confirm if you are using the beta SDK? If so, is this still an issue with the latest 5.75.0-preview?

**Comment by cammurray** [comment]
> Thanks for raising this @cammurray
> 
> Any chance you can confirm if you are using the beta SDK? If so, is this still an issue with the latest 5.75.0-preview?

Hey @andrueastman 

Yep, i'm using the beta SDK and the issue definitely exists on the latest 5.75-0-preview.

**Comment by andrueastman** [comment]
This is caused by missing metadata is missing the required path for the model to be generated. Related to metadata issue at https://github.com/microsoftgraph/msgraph-metadata/issues/531

**Comment by cloudhal** [comment]
I have a very similar issue since around 5.48 using PageIterator with UserCollectionResponse, wonder if this is the same issue? Not an issue if I don't have trimming enabled.

**Comment by andrueastman** [comment]
> I have a very similar issue since around 5.48 using PageIterator with UserCollectionResponse, wonder if this is the same issue? Not an issue if I don't have trimming enabled.

I believe this would be a different issue as the issue here is that the model is not generated in the source due to the input metadata missing it. UserCollectionResponse model however exists in the source at https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/3c7d00ea9c8907fc8649a870c12426aa2fd8a611/src/Microsoft.Graph/Generated/Models/UserCollectionResponse.cs#L10

Any chance you can create a separate issue outlining the issue you are facing?

**Comment by cloudhal** [comment]
> > I have a very similar issue since around 5.48 using PageIterator with UserCollectionResponse, wonder if this is the same issue? Not an issue if I don't have trimming enabled.
> 
> 
> 
> I believe this would be a different issue as the issue here is that the model is not generated in the source due to the input metadata missing it. UserCollectionResponse model however exists in the source at https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/3c7d00ea9c8907fc8649a870c12426aa2fd8a611/src/Microsoft.Graph/Generated/Models/UserCollectionResponse.cs#L10
> 
> 
> 
> Any chance you can create a separate issue outlining the issue you are facing? 

Yes I can, although just realised it's using .Net 7 so I need to upgrade it, may be why it's not working?
